contracts/src/v0.8/dev/ocr2dr/OCR2DRRegistry.sol

```diff
@@ -0,0 +1,783 @@
+
+    s_requestCommitments[requestId] = Commitment(
+      billing,
+      msg.sender,
+      OCR2DRBillableInterface(msg.sender).getRequiredFee(data, billing),
+      getRequiredFee(data, billing)
+    );
+
+    emit BillingStart(msg.sender, requestId, billing.subscriptionId, billing.gasLimit, billing.client);
+    s_consumers[billing.client][billing.subscriptionId] = nonce;
+    return requestId;
+  }
+
+  /**
+   * @inheritdoc OCR2DRRegistryInterface
+   */
+  function getCommitment(bytes32 requestId)
+    external
+    view
+    override
+    returns (
+      address,
+      uint64,
+      uint32
+    )
+  {
+    Commitment memory commitment = s_requestCommitments[requestId];
+    return (commitment.billing.client, commitment.billing.subscriptionId, commitment.billing.gasLimit);
+  }
+
+  function computeRequestId(
+    address don,
+    address client,
+    uint64 subId,
+    uint64 nonce
+  ) private pure returns (bytes32, uint256) {
+    uint256 preSeed = uint256(keccak256(abi.encode(don, client, subId, nonce)));
+    return (keccak256(abi.encode(don, preSeed)), preSeed);
+  }
+
+  /**
+   * @dev calls target address with exactly gasAmount gas and data as calldata
+   * or reverts if at least gasAmount gas is not available.
+   */
+  function callWithExactGas(
+    uint256 gasAmount,
+    address target,
+    bytes memory data
+  ) private returns (bool success) {
+    // solhint-disable-next-line no-inline-assembly
+    assembly {
+      let g := gas()
+      // Compute g -= GAS_FOR_CALL_EXACT_CHECK and check for underflow
+      // The gas actually passed to the callee is min(gasAmount, 63//64*gas available).
+      // We want to ensure that we revert if gasAmount >  63//64*gas available
+      // as we do not want to provide them with less, however that check itself costs
+      // gas.  GAS_FOR_CALL_EXACT_CHECK ensures we have at least enough gas to be able
+      // to revert if gasAmount >  63//64*gas available.
+      if lt(g, GAS_FOR_CALL_EXACT_CHECK) {
+        revert(0, 0)
+      }
+      g := sub(g, GAS_FOR_CALL_EXACT_CHECK)
+      // if g - g//64 <= gasAmount, revert
+      // (we subtract g//64 because of EIP-150)
+      if iszero(gt(sub(g, div(g, 64)), gasAmount)) {
+        revert(0, 0)
+      }
+      // solidity calls check that a contract actually exists at the destination, so we do the same
+      if iszero(extcodesize(target)) {
+        revert(0, 0)
+      }
+      // call and return whether we succeeded. ignore return data
+      // call(gas,addr,value,argsOffset,argsLength,retOffset,retLength)
+      success := call(gasAmount, target, 0, add(data, 0x20), mload(data), 0, 0)
+    }
+    return success;
+  }
+
+  /**
+   * @inheritdoc OCR2DRRegistryInterface
+   */
+  function concludeBilling(
+    bytes32 requestId,
+    bytes calldata response,
+    bytes calldata err,
+    address transmitter,
+    address[MAX_NUM_ORACLES] memory, /* signers */
```

Comment:
If we don't use signers, shall not we simply remove this param?

Reply:
+1